client/packages/programs/src/JsonForms/components/EncounterProgramEvent.tsx

```diff
@@ -89,19 +93,43 @@ const UIComponent = (props: ControlProps) => {
   }
 
   return (
-    <DetailInputWithLabelRow
-      label={label}
-      inputProps={{
-        value: event?.data ?? '',
-        disabled: true,
-        required: props.required,
-        sx: DefaultFormRowSx,
-        error: !!errors,
-        helperText: errors,
-      }}
-      labelWidthPercentage={FORM_LABEL_WIDTH}
-      inputAlignment={'start'}
-    />
+    <>
+      {options?.displayType && options?.displayType === 'number' ? (
```

Comment:
if you add to the style constant as suggested in the previous PR, then perhaps this could be refactored so that only the `Input` prop differs between the cases here